File: corporate-reporting/src/types/FinancialRiskMitigationAction.ts
import { IFinancialRisk } from "./FinancialRisk";
import { IFinancialRiskMitigationActionUpdate } from "./FinancialRiskMitigationActionUpdate";
import { IRiskMitigationAction, RiskMitigationAction } from "./RiskMitigationAction";

export interface IFinancialRiskMitigationAction extends IRiskMitigationAction {
    FinancialRisk?: IFinancialRisk;
    FinancialRiskMitigationActionUpdates?: IFinancialRiskMitigationActionUpdate[];
}

export class FinancialRiskMitigationAction extends RiskMitigationAction implements IFinancialRiskMitigationAction { }

File: ORB.Core/Models/FinancialRiskMitigationAction.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class FinancialRiskMitigationAction : RiskMitigationAction
    {
        public FinancialRiskMitigationAction() : base()
        {
            FinancialRiskRiskMitigationActions = new HashSet<FinancialRiskRiskMitigationAction>();
            FinancialRiskMitigationActionUpdates = new HashSet<FinancialRiskMitigationActionUpdate>();
        }

        public virtual FinancialRisk FinancialRisk { get; set; }
        public virtual ICollection<FinancialRiskRiskMitigationAction> FinancialRiskRiskMitigationActions { get; set; }
        public virtual ICollection<FinancialRiskMitigationActionUpdate> FinancialRiskMitigationActionUpdates { get; set; }
    }
}
